### PR TITLE
Add notes from codebase overview session

### PR DESCRIPTION
**Pull request by scottyschup**
## Changelog
Added a doc to help new devs get started

### CODE REVIEW

File: docs/codebase_overview.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+If you get stuck at any point of the install process, ping Scott or Nicole on Slack or by email, or create an issue on this Github repo. 
+
+### The Rails Way™
+Rails has two major principles:
+* **Don't Repeat Yourself (aka DRY)**  
+When you can re-use code, you should re-use code. Writing DRY code makes the codebase more maintainable, more extensible, and less buggy.
+* **Convention** Over **Configuration (or, Rails is Opinionated)**  
+This just means that rather than manually configuring every little detail, Rails generates a lot of boilerplate code with reasonable defaults. It also means that when developing in Rails, there is usually a right way and a not-so-right way to do things. Learning `The Rails Way™` of doing things will make your life easier, happier, and more fulfilling.  
+That said, there are ways of manually configuring nearly everything, but if you find yourself doing this a lot, you're probably not doing it The Rails Way™ and should re-evaluate some of your life choices.
```
Comment (BintLopez) [inline docs/codebase_overview.md:26]: I know you and your tone and hearing you say this in your voice sounds friendly and funny. But.... I don't think that's gonna come through in text and the "re-evaluate" some of your life choices could come off as a bit offputting/judgmental/unfriendly for folks new to the group, rails, or the repo.

File: docs/codebase_overview.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+(And btdubs, [RailsBridge](https://railsbridge.org) is just and awesome organization in general! And they have other great [tutorials and docs](https://docs.railsbridge.org/docs/) too! You should check them out.)
+
+If you get stuck at any point of the install process, ping Scott or Nicole on Slack or by email, or create an issue on this Github repo. 
+
+### The Rails Way™
+Rails has two major principles:
+* **Don't Repeat Yourself (aka DRY)**  
+When you can re-use code, you should re-use code. Writing DRY code makes the codebase more maintainable, more extensible, and less buggy.
+* **Convention** Over **Configuration (or, Rails is Opinionated)**  
+This just means that rather than manually configuring every little detail, Rails generates a lot of boilerplate code with reasonable defaults. It also means that when developing in Rails, there is usually a right way and a not-so-right way to do things. Learning `The Rails Way™` of doing things will make your life easier, happier, and more fulfilling.
```
Comment (BintLopez) [inline docs/codebase_overview.md:25]: We should give a resource to how to know what the rails way is if we're going to say this

Comment (scottyschup) [reply docs/codebase_overview.md:25]: I was thinking that's what the `Setup and first steps` section was—a list of official and/or introductory resources. I could remove the trademarks if you think that could be confusing to people, like they're looking for a book or software called `The Rails Way™`. Or I could make those links to the Rails guide listed at the top of that section.

**Review by BintLopez — APPROVED** [review]
🦑

**Pull request merged**